app/src/test/java/org/simple/clinic/security/pin/PinEntryEffectHandlerTest.kt

```diff
@@ -0,0 +1,58 @@
+package org.simple.clinic.security.pin
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.verifyNoMoreInteractions
+import org.junit.After
+import org.junit.Test
+import org.simple.clinic.mobius.EffectHandlerTestCase
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
+
+class PinEntryEffectHandlerTest {
+
+  private val uiActions = mock<UiActions>()
+
+  private val effectHandler = PinEntryEffectHandler(
+      bruteForceProtection = mock(),
+      schedulersProvider = TrampolineSchedulersProvider(),
+      uiActions = uiActions,
+      pinVerificationMethod = mock()
+  )
+
+  private val testCase = EffectHandlerTestCase(effectHandler.build())
+
+  @After
+  fun tearDown() {
+    testCase.dispose()
+  }
+
+  @Test
+  fun `when the show network error effect is received, the network error must be shown`() {
+    // when
+    testCase.dispatch(ShowNetworkError)
+
+    // then
+    verify(uiActions).showNetworkError()
```

Comment:
Should we also assert that there are no out going events?

Reply:
Good point, I'll do this.